perceval/backends/core/googlehits.py

```diff
@@ -0,0 +1,251 @@
+
+        logger.info("Fetch process completed")
+
+    @classmethod
+    def has_archiving(cls):
+        """Returns whether it supports archiving items on the fetch process.
+
+        :returns: this backend supports items archive
+        """
+        return True
+
+    @classmethod
+    def has_resuming(cls):
+        """Returns whether it supports to resume the fetch process.
+
+        :returns: this backend supports items resuming
+        """
+        return True
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier from a GoogleHit item."""
+
+        return item['id']
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts the update time from a GoogleHit item.
+
+        The timestamp is based on the current time when the hit was extracted.
+        This field is not part of the data provided by Google API. It is added
+        by this backend.
+
+        :param item: item generated by the backend
+
+        :returns: a UNIX timestamp
+        """
+        return item['fetched_on']
+
+    @staticmethod
+    def metadata_category(item):
+        """Extracts the category from a GoogleHit item.
+
+        This backend only generates one type of item which is
+        'google_hits'.
+        """
+        return CATEGORY_HITS
+
+    def _init_client(self, from_archive=False):
+        """Init client"""
+
+        return GoogleHitsClient(self.sleep_time, self.max_retries,
+                                archive=self.archive, from_archive=from_archive)
+
+    @staticmethod
+    def uuid(*args):
+        """Generate a UUID based on the given parameters."""
+
+        s = '-'.join(args)
+
+        sha1 = hashlib.sha1(s.encode('utf-8', errors='surrogateescape'))
+        uuid_sha1 = sha1.hexdigest()
+
+        return uuid_sha1
+
+
+class GoogleHitsClient(HttpClient):
+    """GoogleHits API client.
+
+    Client for fetching hits data from Google API.
+
+    :param sleep_time: time to sleep in case
+        of connection problems
+    :param max_retries: number of max retries to a data source
+        before raising a RetryError exception
+    :param archive: an archive to store/read fetched data
+    :param from_archive: it tells whether to write/read the archive
+    """
+    def __init__(self, sleep_time=DEFAULT_SLEEP_TIME, max_retries=MAX_RETRIES,
+                 archive=None, from_archive=False):
+        super().__init__(GOOGLE_SEARCH_URL, extra_status_forcelist=[429],
+                         sleep_time=sleep_time, max_retries=max_retries,
+                         archive=archive, from_archive=from_archive)
+
+    def hits(self, keywords):
+        """Fetch information about a list of keywords."""
+
+        for keyword in keywords:
+            logger.info("Fetching hits for '%s'", keyword)
+            params = {'q': keyword}
+
+            # Make the request
+            req = self.fetch(GOOGLE_SEARCH_URL, payload=params)
+
+            # Create the soup and get the desired div
```

Comment:
Is not better to move the parsing code to the backend as we have in other backends?